src/System.Threading.Tasks.Channels/System/IO/Channels/ChannelReader.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Channels
+{
+    /// <summary>
+    /// Provides a base class for reading from a channel.
+    /// </summary>
+    /// <typeparam name="T">Specifies the type of data that may be read from the channel.</typeparam>
+    public abstract class ChannelReader<T>
+    {
+        /// <summary>
+        /// Gets a <see cref="Task"/> that completes when no more data will ever
+        /// be available to be read from this channel.
+        /// </summary>
+        public virtual Task Completion => ChannelUtilities.s_neverCompletingTask;
```

Comment:
Should it be called "Completed"?

Reply:
To me personally "Completed" makes it sound like it's a Boolean.  But I don't have a strong preference.

Reply:
Boolean property would be called "IsCompleted", wouldn't it?

Reply:
That's what we'd name it... I'm just saying I see "Completed" and think "bool".

Reply:
OK, your call.